### ISSUE TITLE
The comment of the example in [dcl.init.general] p16.6.1 is wrong CWG2612

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#5488]
[dcl.init.general] p16.6.1
> [Example [2](https://eel.is/c++draft/dcl.init#general-example-2): T x = T(T(T())); calls the T default constructor to initialize x[.](https://eel.is/c++draft/dcl.init#general-16.6.1.sentence-2) — end example]

The result object `x` is not definitely initialized by the default constructor. The result object is actually value-initialized.
> If the initializer is (), the object is value-initialized[.](https://eel.is/c++draft/dcl.init.general#16.4.sentence-1)  

In this case, the result object will be zero-initialized if `T` has a **trivial default constructor**. This point can be shown by this example. 
````cpp
struct A{
    int a;
};

int main(){
   A a = A(A(A()));  // #1
   A b;
   std::cout<<"a.a: "<< a.a<<std::endl;
   std::cout<< "b.a: "<< b.a<<std::endl;
}
````
The result printed by [Clang](https://godbolt.org/z/Er9MxE1Md) is  
````
a.a: 0   // the subobject is zero-initialized
b.a: -1215191136  // indeterminate value
````
If the default constructor were called for `#1`, the subobject `A::a` would be uninitialized.

The comment should just say `x` is value-initialized.

**Comment by frederick-vs-ja** [comment]
This should have been fixed by #5984.

**Comment by jensmaurer** [comment]
Fixed by CWG2612.
